### PR TITLE
Add nav prop to Dropdown

### PR DESCRIPTION
**Pull request by supergibbs**
Consolidate NavDropdown with Dropdown
Move NavDropdown tests to Dropdown
Remove Uncontrolled NavDropdown and tests
Update docs

https://github.com/reactstrap/reactstrap/issues/326#issuecomment-279780826

### CODE REVIEW

**Comment by supergibbs** [comment]
I updated the tests but I can't seem to run them locally. I get the following:

```
➜  reactstrap git:(consolidate-navdropdown) ✗ npm test 

> reactstrap@5.0.0-alpha.3 test /Users/jmandel/Development/supergibbs/reactstrap
> cross-env BABEL_ENV=test react-scripts test --env=jsdom

2017-10-14 12:48 node[8340] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
2017-10-14 12:48 node[8340] (FSEvents.framework) FSEventStreamStart: register_with_server: ERROR: f2d_register_rpc() => (null) (-22)
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: Error watching file for changes: EMFILE
    at exports._errnoException (util.js:1020:11)
    at FSEvent.FSWatcher._handle.onchange (fs.js:1421:9)
npm ERR! Test failed.  See above for more details.
```

Any ideas? Running NodeJS v6.11.4, npm v5.5.1 and a fresh `npm install` after removing node_modules/lock file

**Comment by TheSharpieOne** [comment]
I get that on my work Mac as well. I am able run coverage which does the test but does not watch the files. Takes a little longer to run, but at least it works.

File: package.json [hunk 1]
```diff
@@ -64,6 +64,7 @@
     "Jason Sturges <jason@jasonsturges.com> (https://github.com/jasonsturges)",
     "Jean-Elie Barjonet (https://github.com/jebarjonet)",
     "Jeff Francisco (https://github.com/j-francisco)",
+    "Jesse Mandel (https://github.com/supergibbs)",
```
Comment (TheSharpieOne) [inline package.json]: I think this is generated, but eh.

File: src/Uncontrolled.js [hunk 2]
```diff
@@ -44,13 +42,11 @@ Object.keys(components).forEach((key) => {
...
 const UncontrolledButtonDropdown = components.UncontrolledButtonDropdown;
 const UncontrolledDropdown = components.UncontrolledDropdown;
-const UncontrolledNavDropdown = components.UncontrolledNavDropdown;
 const UncontrolledTooltip = components.UncontrolledTooltip;
 
 export {
   UncontrolledAlert,
   UncontrolledButtonDropdown,
   UncontrolledDropdown,
-  UncontrolledNavDropdown,
```
Comment (TheSharpieOne) [inline src/Uncontrolled.js]: Would be nice to deprecate this the same as `NavDropdown` so that it is not a breaking change.

File: src/index.js [hunk 3]
```diff
@@ -6,7 +6,6 @@ import NavbarBrand from './NavbarBrand';
 import NavbarToggler from './NavbarToggler';
 import Nav from './Nav';
 import NavItem from './NavItem';
-import NavDropdown from './NavDropdown';
```
Comment (TheSharpieOne) [inline src/index.js]: You still need to import this and export it so that the deprecated component can still be used (and give people the warning).

**Comment by supergibbs** [comment]
I am happy to make those changes but I am traveling for a couple weeks so
if you want to merge it in before feel free to work on it.

**Comment by supergibbs** [comment]
@TheSharpieOne I'm back and made those changes! Let me know if there is anything else

**Comment by TheSharpieOne** [comment]
Thanks @supergibbs for making those changes. I was hoping to get around to it but have been too swamped.

**Pull request merged**